mlrun/runtimes/function.py

```diff
@@ -41,39 +41,40 @@
 default_max_replicas = 4
 
 
-def new_model_server(
+def new_v1_model_server(
     name,
     model_class: str,
     models: dict = None,
     filename="",
     protocol="",
     image="",
     endpoint="",
-    explainer=False,
     workers=8,
     canary=None,
-    handler=None,
 ):
     f = RemoteRuntime()
     if not image:
         name, spec, code = nuclio.build_file(
             filename, name=name, handler=serving_handler, kind="serving"
         )
         f.spec.base_spec = spec
-    elif handler:
-        f.spec.function_handler = handler
 
     f.metadata.name = name
-    f.serving(
-        models,
-        model_class,
-        protocol,
-        image=image,
-        endpoint=endpoint,
-        explainer=explainer,
-        workers=workers,
-        canary=canary,
-    )
+
+    if models:
+        for k, v in models.items():
+            f.set_env("SERVING_MODEL_{}".format(k), v)
+
+    if protocol:
+        f.set_env("TRANSPORT_PROTOCOL", protocol)
+    if model_class:
+        f.set_env("MODEL_CLASS", model_class)
+    f.with_http(workers, host=endpoint, canary=canary)
+    f.spec.function_kind = "serving"
```

Comment:
this is the subkind (not kind), it was the same code before

Reply:
And this is exactly why use constants, cause if I get you correctly, `"serving"` not only represents a function kind but also a function subkind, by using constants I know what you're referring to.
Of course if it's subkind, then it would make much more sense to call the attribute... subkind, but I guess the answer to that is that it is already like that ?